File: inspect_dom.py
import asyncio
from playwright.sync_api import sync_playwright

def inspect_page_headers():
    with sync_playwright() as p:
        # Launch browser
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()
        
        # Navigate to the Schedule page
        page.goto('http://localhost:1234/schedule')
        page.wait_for_load_state('networkidle')
        
        print("=== INSPECTING SCHEDULE PAGE ===\n")
        
        # Find the h1 element
        h1_element = page.query_selector('h1')
        if h1_element:
            h1_text = h1_element.text_content()
            print(f"H1 Text: {h1_text}")
            
            # Get all children of the h1's parent
            children_info = page.evaluate('''() => {
                const h1 = document.querySelector('h1');
                if (!h1 || !h1.parentElement) return [];
                
                return Array.from(h1.parentElement.children).map(child => ({
                    tagName: child.tagName,
                    className: child.className,
                    textContent: child.textContent?.substring(0, 100)
                }));
            }''')
            
            print("\n=== H1 PARENT'S CHILDREN ===")
            for i, child in enumerate(children_info):
                print(f"\nChild {i}:")
                print(f"  Tag: {child['tagName']}")
                print(f"  Class: {child['className']}")
                print(f"  Text: {child['textContent']}...")
        
        # Look for subtitle text
        subtitle_search = "Gerencie todos os agendamentos"
        elements_with_subtitle = page.query_selector_all(f'text="{subtitle_search}"')
        
        print(f"\n=== FOUND {len(elements_with_subtitle)} ELEMENTS WITH SUBTITLE TEXT ===")
        for i, element in enumerate(elements_with_subtitle):
            tag = element.evaluate('(el) => el.tagName')
            classes = element.evaluate('(el) => el.className')
            parent_tag = element.evaluate('(el) => el.parentElement.tagName')
            
            print(f"\nElement {i}:")
            print(f"  Tag: {tag}")
            print(f"  Classes: {classes}")
            print(f"  Parent Tag: {parent_tag}")
            
            # Get surrounding HTML
            surrounding = element.evaluate('''(el) => {
                const parent = el.parentElement;
                if (!parent) return '';
                return parent.innerHTML.substring(0, 300);
            }''')
            print(f"  Surrounding HTML: {surrounding}...")
        
        # Check for any p tags with text-muted-foreground right after h1
        suspicious_elements = page.evaluate('''() => {
            const results = [];
            const h1s = document.querySelectorAll('h1');
            
            h1s.forEach(h1 => {
                let next = h1.nextElementSibling;
                while (next && next.tagName !== 'H1' && next.tagName !== 'DIV') {
                    if (next.tagName === 'P' || next.className.includes('text-muted')) {
                        results.push({
                            h1Text: h1.textContent,
                            nextTag: next.tagName,
                            nextClass: next.className,
                            nextText: next.textContent?.substring(0, 100)
                        });
                    }
                    next = next.nextElementSibling;
                }
            });
            
            return results;
        }''')
        
        if suspicious_elements:
            print("\n=== SUSPICIOUS ELEMENTS AFTER H1 ===")
            for item in suspicious_elements:
                print(f"\nAfter H1 '{item['h1Text']}':")
                print(f"  Found: <{item['nextTag']} class='{item['nextClass']}'>{item['nextText']}</...>")
        
        browser.close()

if __name__ == "__main__":
    inspect_page_headers()

File: inspect-dom.mjs
import { chromium } from 'playwright';

const browser = await chromium.launch({ headless: true });
const page = await browser.newPage();

console.log('\n🔍 === INSPECTING SCHEDULE PAGE ===\n');

await page.goto('http://localhost:1234/schedule');
await page.waitForLoadState('networkidle');

// Find h1 and its siblings
const h1Info = await page.evaluate(() => {
  const h1 = document.querySelector('h1');
  if (!h1) return null;
  
  const parent = h1.parentElement;
  const siblings = Array.from(parent.children);
  
  return {
    h1Text: h1.textContent,
    h1Classes: h1.className,
    parentTag: parent.tagName,
    parentClasses: parent.className,
    siblings: siblings.map(el => ({
      tag: el.tagName,
      classes: el.className,
      text: el.textContent?.substring(0, 100),
      isH1: el === h1
    }))
  };
});

if (h1Info) {
  console.log(`📌 H1 Text: "${h1Info.h1Text}"`);
  console.log(`📦 Parent: <${h1Info.parentTag} class="${h1Info.parentClasses}">`);
  console.log('\n📋 Siblings in same container:');
  
  h1Info.siblings.forEach((sibling, i) => {
    const marker = sibling.isH1 ? '👉' : '  ';
    console.log(`${marker} [${i}] <${sibling.tag} class="${sibling.classes}">`);
    console.log(`     Text: "${sibling.text}"`);
  });
}

// Search for the subtitle text
console.log('\n🔍 === SEARCHING FOR SUBTITLE TEXT ===\n');

const subtitleElements = await page.$$eval(
  ':has-text("Gerencie todos os agendamentos")',
  elements => elements.map(el => ({
    tag: el.tagName,
    classes: el.className,
    id: el.id,
    text: el.textContent?.substring(0, 150),
    parentTag: el.parentElement?.tagName,
    parentClasses: el.parentElement?.className,
    prevSiblingIsH1: el.previousElementSibling?.tagName === 'H1',
    path: (() => {
      let current = el;
      let path = [];
      while (current && current.tagName !== 'BODY') {
        path.push(`${current.tagName}.${current.className || 'no-class'}`);
        current = current.parentElement;
      }
      return path.slice(0, 5).join(' > ');
    })()
  }))
);

if (subtitleElements.length > 0) {
  console.log(`⚠️  Found ${subtitleElements.length} elements with subtitle text!\n`);
  
  subtitleElements.forEach((el, i) => {
    console.log(`📍 Element ${i + 1}:`);
    console.log(`   Tag: <${el.tag} class="${el.classes}">`);
    console.log(`   Text: "${el.text}"`);
    console.log(`   Parent: <${el.parentTag} class="${el.parentClasses}">`);
    if (el.prevSiblingIsH1) {
      console.log(`   ⚠️  PREVIOUS SIBLING IS H1!`);
    }
    console.log(`   Path: ${el.path}`);
    console.log('');
  });
} else {
  console.log('✅ No subtitle text found in DOM!');
}

// Check for p.text-muted-foreground elements
console.log('\n🔍 === CHECKING FOR p.text-muted-foreground ===\n');

const mutedParagraphs = await page.$$eval(
  'p.text-muted-foreground',
  elements => elements.slice(0, 5).map(el => ({
    text: el.textContent?.substring(0, 100),
    prevSiblingTag: el.previousElementSibling?.tagName,
    prevSiblingText: el.previousElementSibling?.textContent?.substring(0, 50)
  }))
);

if (mutedParagraphs.length > 0) {
  console.log(`Found ${mutedParagraphs.length} p.text-muted-foreground elements:\n`);
  mutedParagraphs.forEach((el, i) => {
    console.log(`[${i + 1}] Text: "${el.text}"`);
    if (el.prevSiblingTag === 'H1') {
      console.log(`    ⚠️  Previous sibling is H1: "${el.prevSiblingText}"`);
    }
  });
}

await browser.close();
console.log('\n✅ Inspection complete!\n');